File: spring-boot-ecommerce/src/main/java/com/sougoumay/ecommerce/dto/Purchase.java
package com.sougoumay.ecommerce.dto;

import com.sougoumay.ecommerce.entity.Address;
import com.sougoumay.ecommerce.entity.Customer;
import com.sougoumay.ecommerce.entity.Order;
import com.sougoumay.ecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;


}

File: angular-ecommerce/src/app/modeles/purchase/purchase.ts
import {Customer} from "../customer/customer";
import {Address} from "../address/address";
import {Order} from "../order/order";
import {OrderItem} from "../order/item/order-item";

export class Purchase {
  // customer : Customer;
  // shippingAddress : Address;
  // billingAddress : Address;
  // order : Order;
  // orderItems : OrderItem;

  constructor(
    public customer : Customer,
    public shippingAddress : Address,
    public billingAddress : Address,
    public order : Order,
    public orderItems : OrderItem[]
  ) {
  }

  // public setShippingAddress(shippingAddress) {
  //   this.shippingAddress.state = shippingAddress.state;
  // }
}
